src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2025, Hewlett Packard Enterprise Development LP */
+
+/*
+ * util_sds_check.c -- unit test for shutdown_state_check function
+ */
+
+#include "util_sds_check.h"
+
+FUNC_MOCK(shutdown_state_reinit, void, struct shutdown_state *curr_sds,
+	struct shutdown_state *pool_sds, struct pool_replica *rep)
+FUNC_MOCK_RUN_DEFAULT {
+	ASSERTeq(curr_sds, &Curr_sds);
+	ASSERTeq(pool_sds, &Pool_sds);
+	ASSERTeq(rep, &Rep);
+}
+FUNC_MOCK_END
+
+/* Test with curr_sds having non-zero values and pool_sds being zeroed out. */
+static int
+test_dirty_clear()
+{
+	SET_SDS(Pool_sds, SDS_ZERO);
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	ASSERTeq(ret, 0);
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/*
+ * Test with curr_sds and pool_sds having the same values
+ * except for the dirty field.
+ */
+static int
+test_dirty_set()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID, DIRTY_SET, CHECKSUM));
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
```

Comment:
```suggestion
	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID, DIRTY_SET, CHECKSUM));
	shutdown_state_checksum(&Pool_sds, &Rep);
	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
```

Otherwise, the SDS is reinitialized but for a different reason. 🙃

Reply:
Done.